taichi/backends/vulkan/aot_module_builder_impl.cpp

```diff
@@ -0,0 +1,75 @@
+#include "taichi/backends/vulkan/aot_module_builder_impl.h"
+
+#include <fstream>
+#include <type_traits>
+
+#include "taichi/backends/vulkan/codegen_vulkan.h"
+
+namespace taichi {
+namespace lang {
+namespace vulkan {
+
+AotModuleBuilderImpl::AotModuleBuilderImpl(
+    VkRuntime *runtime,
+    const std::vector<CompiledSNodeStructs> &compiled_structs)
+    : compiled_structs_(compiled_structs), runtime_(runtime) {
+}
+
+void AotModuleBuilderImpl::write_spv_file(
+    const std::string &output_dir,
+    const std::string &filename,
+    const TaskAttributes &k,
+    const std::vector<uint32_t> &source_code) const {
+  const std::string spv_path =
+      fmt::format("{}/{}_{}.spv", output_dir, filename, k.name);
```

Comment:
btw we recently found out that ignoring the filename prefix is probably more convenient (https://github.com/taichi-dev/taichi/pull/3648), I'm curious if you want the same for vulkan backend ;)

Reply:
Definitely! I pushed the same changes to this PR, that would avoid to do multiple one :). I tested it and seems to work as well.